src/main/java/com/sap/broker/budgie/configuration/behavior/BehaviorEngine.java

```diff
@@ -0,0 +1,155 @@
+package com.sap.broker.budgie.configuration.behavior;
+
+import com.sap.broker.budgie.configuration.ApplicationConfiguration;
+import com.sap.broker.budgie.domain.Plan;
+import com.sap.broker.budgie.domain.Service;
+import com.sap.broker.budgie.domain.ServiceInstance;
+import org.springframework.stereotype.Component;
+
+import javax.inject.Inject;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.BiPredicate;
+import java.util.stream.Stream;
+
+@Component
+public class BehaviorEngine {
+
+    private ApplicationConfiguration appConfiguration;
+
+    private Map<String, BehaviorConfiguration> configurations = new HashMap<>();
+
+    @Inject
+    public BehaviorEngine(ApplicationConfiguration configuration) {
+        this.appConfiguration = configuration;
+    }
+
+    public synchronized void addConfiguration(String id, BehaviorConfiguration configuration) {
+        configurations.put(id, configuration);
+    }
+
+    public synchronized BehaviorConfiguration removeConfiguration(String id) {
+        return configurations.remove(id);
+    }
+
+    public synchronized BehaviorConfiguration getConfiguration(String id) {
+        return configurations.get(id);
+    }
+
+    public synchronized Map<String, BehaviorConfiguration> getConfigurations() {
+        return Collections.unmodifiableMap(configurations);
+    }
+
+    public Integer getDuration(String id) {
+        BehaviorConfiguration configuration = getConfiguration(id);
+        if (isAsync(id)) {
+            return configuration.getAsyncDuration();
+        }
+        if (configuration != null && configuration.getSyncDuration() != null) {
```

Comment:
Could we make isSync method which is consistent with the other methods isAsync method?